boards/arm/thingy91_nrf9160/Kconfig.board

```diff
@@ -12,4 +12,36 @@ config BOARD_THINGY91_NRF9160
 config BOARD_THINGY91_NRF9160_NS
 	bool "nRF9160 THINGY91 non-secure"
 
+if BOARD_THINGY91_NRF9160_NS
+
+choice
+	prompt "Thingy:91 partition layout"
+	default THINGY91_STATIC_PARTITIONS_SECURE_BOOT if SECURE_BOOT
+	default THINGY91_STATIC_PARTITIONS_LWM2M_CARRIER if LWM2M_CARRIER
+	default THINGY91_STATIC_PARTITIONS_FACTORY
+
+config THINGY91_STATIC_PARTITIONS_FACTORY
+	bool "Factory Thingy:91 partition layout"
+	help
+	   The default Thingy:91 partition layout used in the factory firmware. This ensures
+	   firmware updates are compatible with Thingy:91 when flashing firmware over USB or over
+	   the air.
+
+config THINGY91_STATIC_PARTITIONS_SECURE_BOOT
```

Comment:
question, should this has a `depends on SECURE_BOOT` ?
Cause otherwise the choice will be sticky after first CMake invocation, in such a way that if someone does an initial build and later uses menuconfig to enable / disable secure boot, then the partition layout choice will not change.

Thingy is a bit special, so the current behavior (stickiness) might be what you want.